endpoints-framework/src/main/java/com/google/api/server/spi/ServiceException.java

```diff
@@ -153,6 +153,21 @@ public ServiceException putExtraField(String fieldName, Number value) {
     return putExtraFieldInternal(fieldName, value);
   }
 
+  /**
+   * Associates to this exception an extra field as a field name/value pair. If a field
+   * with the same name was previously set, the old value is replaced by the specified
+   * value.<br>
+   * This unsafe version accepts any POJO as is: nor defensive copy nor conversion are made.
+   * So {@code value} should not be modified or reused after the call of this method.
+   * @return this
+   * @throws NullPointerException if {@code fieldName} is {@code null}.
+   * @throws IllegalArgumentException if {@code fieldName} is one of the reserved field
+   *         names {@link #EXTRA_FIELDS_RESERVED_NAMES}.
+   */
+  protected ServiceException putExtraFieldUnsafe(String fieldName, Object value) {
```

Comment:
Is there any particular reason to set this method as `protected` while the other flavors of `putExtraField` are public ? This one fulfills exactly the same need as the other.
Same way, suffixing its name by `unsafe` is really puzzling since it has exactly the same purpose of the others. Plus, this suffix is not really invitation to use it). I would set its signature as the following : 
```suggestion protected ServiceException putExtraFieldUnsafe(String fieldName, Object value) {
public ServiceException putExtraField(String fieldName, Object value) {
```